packages/@lwc/engine-core/src/framework/base-lightning-element.ts

```diff
@@ -342,50 +324,62 @@ LightningElement.prototype = {
     ): void {
         const vm = getAssociatedVM(this);
         const { elm } = vm;
+        const { removeEventListener } = vm.renderer;
 
         const wrappedListener = getWrappedComponentsListener(vm, listener);
         removeEventListener(elm, type, wrappedListener, options);
     },
 
     hasAttribute(name: string): boolean {
-        const { elm } = getAssociatedVM(this);
+        const vm = getAssociatedVM(this);
+        const { elm } = vm;
+        const { getAttribute } = vm.renderer;
         return !isNull(getAttribute(elm, name));
     },
 
     hasAttributeNS(namespace: string | null, name: string): boolean {
-        const { elm } = getAssociatedVM(this);
+        const vm = getAssociatedVM(this);
+        const { elm } = vm;
+        const { getAttribute } = vm.renderer;
         return !isNull(getAttribute(elm, name, namespace));
     },
 
     removeAttribute(name: string): void {
-        const { elm } = getAssociatedVM(this);
-
+        const vm = getAssociatedVM(this);
+        const { elm } = vm;
+        const { removeAttribute } = vm.renderer;
         unlockAttribute(elm, name);
         removeAttribute(elm, name);
         lockAttribute(elm, name);
     },
 
     removeAttributeNS(namespace: string | null, name: string): void {
-        const { elm } = getAssociatedVM(this);
-
+        const vm = getAssociatedVM(this);
+        const { elm } = vm;
+        const { removeAttribute } = vm.renderer;
```

Comment:
```suggestion
        const { elm, renderer: { removeAttribute } } = getAssociatedVM(this);
```

Just a suggestion for here and other parts of the code. Could be a bit terser.